IntegrationTests/UI.IntegrationTests/UITest.cs

```diff
@@ -33,44 +33,23 @@ void HandleApplicationIdle(object sender, EventArgs e)
             Func<T, Task> runTestAsync)
             where T : Form
         {
-            // If form.Dipose was called by the async test task, closing the window would be done in a strange order.
+            Assert.IsEmpty(Application.OpenForms, $"{Application.OpenForms.Count} open form(s) before test");
+
             T form = null;
             try
             {
-                // Start runTestAsync before calling showForm.
-                // The latter might block until the form is closed, especially if using Application.Run(form).
-
+                // Start runTestAsync before calling showForm, since the latter might block until the form is closed.
+                //
                 // Avoid using ThreadHelper.JoinableTaskFactory for the outermost operation because we don't want the task
                 // tracked by its collection. Otherwise, test code would not be able to wait for pending operations to
                 // complete.
                 var test = ThreadHelper.JoinableTaskContext.Factory.RunAsync(async () =>
                 {
-                    try
-                    {
-                        await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                        await WaitForIdleAsync();
-                        form = Application.OpenForms.OfType<T>().Single();
-                    }
-                    catch (Exception ex)
-                    {
-                        // This exception might not be visible because showForm might not return if the opened form is not closed.
-                        // Log it for the case the "finally" code does not help.
-                        Console.WriteLine($"{nameof(RunForm)}<{typeof(T).FullName}> async test preparation threw {ex.Demystify()}");
-                        throw;
-                    }
```

Comment:
📝 This block was effectively unreachable, so the code is simplified by removing it.

Reply:
You must write long explanations which do not fully convince me (see above).
Every experienced developer will ask the same questions.
It is not even visible that you have good reasons to think this.